pkg/machines/components/create-vm-dialog/pxe-helpers.js

```diff
@@ -0,0 +1,192 @@
+            else
+                return _("Routed Network");
+        } else {
+            if (dev)
+                return cockpit.format('$0 to $1', mode, dev);
+            else
+                return cockpit.format(_("$0 Network"), mode.toUpperCase());
+        }
+    } else {
+        return _("Isolated Network");
+    }
+}
+
+/**
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {object} virtualNetwork - A Virtual Network object.
+ */
+export function getVirtualNetworkPXESupport(virtualNetwork) {
+    if (virtualNetwork.forward && virtualNetwork.forward.mode != 'nat') {
+        return true;
+    }
+
+    return !!virtualNetwork.ip.find(ip => ip.dhcp.bootp);
+}
+
+/**
+ * Returns the first available Network Resource to be used for showing to PXE Network Sources list.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getPXEInitialNetworkSource(nodeDevices, virtualNetworks, connectionName) {
+    if (virtualNetworks.length > 0)
+        return cockpit.format('network=$0', virtualNetworks[0].name);
+
+    let netNodeDevices = filterVirtualBridgesFromNetNodeDevices(
+        getNodeDevicesOfType(nodeDevices, connectionName, 'net'),
+        virtualNetworks
+    );
+
+    if (netNodeDevices.length > 0)
+        return cockpit.format('type=direct,source=$0', netNodeDevices[0].capability.interface);
+}
+
+/**
+ * Returns the Select Entries rows for the PXE Network Sources.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getPXENetworkRows(nodeDevices, virtualNetworks, connectionName) {
+    /* Do not show to the user the libvirt virbrX-nic devices since these are
+     * supposed to be managed through virtual networks
+     */
+    let netNodeDevices = filterVirtualBridgesFromNetNodeDevices(
+        getNodeDevicesOfType(nodeDevices, connectionName, 'net'),
+        virtualNetworks
+    );
+
+    let virtualNetworkRows = virtualNetworks.map(network => {
+        const data = cockpit.format('network=$0', network.name);
+
+        return (
+            <Select.SelectEntry
+                data={data}
+                key={data}>
+                {cockpit.format("$0 $1: $2", _("Virtual Network"), network.name, getVirtualNetworkDescription(network))}
+            </Select.SelectEntry>
+        );
+    });
+
+    let netNodeDevicesRows = netNodeDevices.map(netNodeDevice => {
+        const iface = netNodeDevice.capability.interface;
+        const data = cockpit.format('type=direct,source=$0', iface);
+
+        return (
+            <Select.SelectEntry
+                data={data}
+                key={data}>
```

Comment:
This could now become one line (totally optional, but as you need to push anyway..)

Reply:
Done